### ISSUE TITLE
bug: Failed to run `config` for fzf-lua

### ISSUE THREAD

**Issue by Drewtopia** [LazyVim/LazyVim#3650]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.10.0

### Operating system/version

MacOS 14.5

### Describe the bug

Failed to run `config` for fzf-lua error after updating LazyVim and installing the `editor.fzf` via :LazyExtras

![CleanShot 2024-06-14 at 22 38 05@2x](https://github.com/LazyVim/LazyVim/assets/1256558/7b8012c2-5d78-41f6-a143-8b7bbe7741a0)
![CleanShot 2024-06-14 at 22 54 23@2x](https://github.com/LazyVim/LazyVim/assets/1256558/d7980511-69f5-42f6-8f07-69036369b6d0)

### Steps To Reproduce

1. Update LazyVim (including with `repro.lua`)
2. Install `editor.fzf` via :LazyExtras
3. Restart LazyVim

### Expected Behavior

`fzf-lua` should work as expected.

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  { "LazyVim/LazyVim", import = "lazyvim.plugins" },
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

**Comment by mirsella** [comment]
hello, this seems to not be lazyvim related
there is issue for this on the fzf-lua repo: https://github.com/ibhagwan/fzf-lua/issues?q=failed+to+start+server

you can see this issue on the neovim repo https://github.com/neovim/neovim/issues/20865 about this issue.
tldr: nvim try to use xdg spec but they are not specified on mac, you can manually create the folder nvim is trying to use or change it

**Comment by Drewtopia** [comment]
Thanks for looking, I stumbled across this at the same time.

I'll see what I can do

**Comment by ibhagwan** [comment]
@Drewtopia, once you hopefully find the solution pls post it also to https://github.com/ibhagwan/fzf-lua/issues/1243 so others can benefit from your work :)

**Comment by ibhagwan** [comment]
> @ibhagwan the solution is found no ? it's just the wrong env variable ? $XDG_RUNTIME_DIR ?

That's my belief too but you never know, might be on Windows or some other wild env issue (nix, etc), for example https://github.com/ibhagwan/fzf-lua/issues/1226.

**Comment by mirsella** [comment]
you're right, for this case he said he's on MacOS 14.5 and the issue on the neovim repo was also on macos because of xdg env, so i guess it's the same.
i hope OP will give a update otherwise !

**Comment by Drewtopia** [comment]
Added my update here, hopefully this helps anyone facing similar issues.

https://github.com/ibhagwan/fzf-lua/issues/1243#issuecomment-2168891260
